php/libraries/Visit.class.inc

```diff
@@ -0,0 +1,220 @@
+     */
+    public function getLabel($id)
+    {
+        $DB = Database::singleton();
+
+        $result = $DB->pselect(
+            "SELECT VisitLabel, VisitLegacyLabel
+             FROM visit
+             WHERE VisitID = :id",
+            array('id' => $id)
+        );
+        return $result;
+    }
+
+    /**
+     * Return an array of hash containing all the visit available
+     * for a project
+     *
+     * @param array $param array['projectID'] the ID of the project
+     *                     array['subprojectID'] the ID of the subproject
+     *                     for which the visit should be available
+     *                     if none, all visit
+     *
+     * @throws DatabaseException
+     *
+     * @return array an array of hash with id and both label
+     *               array is of the form
+     *               array(0 => array('VisitID'     => visit ID,
+     *                                'Label'       => 'label',
+     *                                'LegacyLabel' => 'legacy label'))
+     */
+    public static function getList($param = array())
+    {
+        $DB = Database::singleton();
+
+        $where      = "";
+        $queryParam = array();
+        if (isset($param['projectID']) && !is_null($param['projectID'])) {
+            $where = " JOIN visit_project_subproject_rel using(VisitID)
+                       WHERE ProjectID = :projectID";
+            $queryParam['projectID'] = $param['projectID'];
+            if (empty($param['subprojectID'])) {
+                $where .= " AND SubprojectID IS NULL";
+            } else {
+                $where .= " AND SubprojectID = :subprojectID";
+                $queryParam['subprojectID'] = $param['subprojectID'];
+
+            }
+        } elseif (isset($param['subprojectID'])
+            && !is_null($param['subprojectID'])
+        ) {
+            $where = " JOIN visit_project_subproject_rel using(VisitID)
+                       WHERE SubprojectID = :subprojectID AND ProjectID IS NULL";
+            $queryParam['subprojectID'] = $param['subprojectID'];
+        }
+        $result = $DB->pselect(
+            "SELECT VisitID, VisitLabel, VisitLegacyLabel
+             FROM visit" . $where,
+            $queryParam
+        );
+        return $result;
+    }
+
+    /**
+     * Modify the label and legacyLabel of a visit
+     *
+     * @param string $id          the ID of the visit
+     * @param string $label       the front-end label
+     * @param string $legacyLabel the back-end label
+     *
+     * @throws DatabaseException
+     *
+     * @return void
+     */
+    public function setLabel($id, $label, $legacyLabel=null)
+    {
+        $DB = Database::singleton();
+
+        $set = array('VisitLabel' => $label);
+        if (!is_null($legacyLabel)) {
+            $set['VisitLegacyLabel'] = $legacyLabel;
+        }
+        $DB->update("visit", $set, array('VisitID' => $id));
+    }
+
+    /**
+     * Create a new visit and label and legacyLabel
```

Comment:
This sounds like it creates a new visit object, but it has the side-effect of modifying the database.